### PR TITLE
Print status in SHOW CLUSTER REPLICAS

### PR DESCRIPTION
**Pull request by umanwizard**
This is a useful thing to expose to users. Proposed by @ggnall in his observability notion doc.

### Motivation

  * This PR adds a feature that has not yet been specified.

The user should be able to easily see whether a replica is online

### Tips for reviewer

The formatting of the SQL query is from mz.sqlfum.pt

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - `SHOW CLUSTER REPLICAS` now prints the replica status

### CODE REVIEW

File: src/adapter/src/catalog/builtin.rs [hunk 1]
```diff
@@ -2344,17 +2344,34 @@ FROM
...
+                    ON mz_catalog.mz_cluster_replicas.cluster_id = mz_catalog.mz_clusters.id
+                JOIN
+                    (
+                            SELECT
+                                replica_id,
+                                mz_internal.mz_all(
+                                        mz_internal.mz_cluster_replica_statuses.status
+                                        = 'ready'
+                                    )
+                                    AS is_ready
```
Comment (jkosh44) [inline src/adapter/src/catalog/builtin.rs:2368]: As per our design principles, this should just be `ready` (https://www.notion.so/materialize/SQL-design-principles-2e2069797f7c46b8b4c7dd10043afdcd)

**Review by jkosh44 — APPROVED** [review]
Mostly looks good, I had one small nit about the naming. Also before merging can you please
- update the docs for `SHOW CLUSTER REPLICAS`
- Update the tests in `cluster.slt`
- Add the `ready` column to the index `MZ_SHOW_CLUSTER_REPLICAS_IND` in `builtin.rs`

**Comment by umanwizard** [comment]
The failing tests seem to have surfaced a bug which might be unrelated: a `SHOW CLUSTER REPLICAS` immediately following a `CREATE CLUSTER REPLICA` command might not reflect the replica it created.

@jkosh44 is looking in to that now.

**Comment by jkosh44** [comment]
> The failing tests seem to have surfaced a bug which might be unrelated: a `SHOW CLUSTER REPLICAS` immediately following a `CREATE CLUSTER REPLICA` command might not reflect the replica it created.
> 
> @jkosh44 is looking in to that now.

So the issue here is that `MZ_CLUSTER_REPLICA_STATUSES` is not immediately populated when a cluster replica is created. Since we join on that table, the query result will not contain any replicas who's status is unknown.

We should really be populating this table immediately on replica creation, because there's a couple of places where it's assumed that a row exists for all replicas. I can push a commit to this PR to fix that.

**Pull request merged**